cram_3d_world/cram_bullet_reasoning_belief_state/cram-bullet-reasoning-belief-state.asd

```diff
@@ -38,6 +38,7 @@
                cram-common-designators ; for world-state-detecting motion
                cram-common-failures ; for world-state-detecting motion
                cram-bullet-reasoning
+               cram-bullet-reasoning-utilities
```

Comment:
I don't like this dependency. I prefer using the utils only for hacking and high-level application code.
But for the libraries I think we should keep the dependencies to the minimum and just use direct btr stuff.
For moving objects, please use `(setf (pose (btr:object btr:*current-bullet-world* :my-obj-name)) my-new-pose)`. For deleting objects I don't remember the exact `btr` function, but you can either use `(btr:retract btr:object ?world ...` Prolog predicate or see how the `retract` is implemented in `btr` and see which low-level function that predicate is calling.

Reply:
Hi, I'm working towards getting this dependency removed.
I've gotten the move and delete working but my spawn-object (used in tests) is not working. 
Just to see its behavior, I tried copy-pasting the spawn-object definition from btr-utils and that hasn't helped either. Do you know any alternatives where this is used?

Reply:
Hmm, what did you try?
Why not use the stuff from the Bullet tutorial then:
```lisp
CL-USER> (prolog:prolog '(and (btr:bullet-world ?world)
                              (assert (btr:object ?world :mesh mug-1 ((0 0 2) (0 0 0 1))
                                                  :mass 0.2 :color (1 0 0) :mesh :mug))))
```

http://cram-system.org/tutorials/intermediate/bullet_world#spawn_and_change_objects_in_the_world

You have to provide a mass but not the color.
Let me know if that doesn't work.

Reply:
I removed the dependency in the latest change. 1 test case now breaks due to some issue, which I haven't figured out yet. The object in that particular test alone acts as if there is no floor and keeps falling endlessly. I have commented that test out alone and investigating it meanwhile.